src/main/java/jenkins/scm/impl/avatars/AvatarCache.java

```diff
@@ -580,59 +603,23 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
      * A task to fetch an image from a remote URL.
      */
     private static class FetchImage implements Callable<CacheEntry> {
-        private final String url;
+        private final AvatarCacheSource source;
```

Comment:
It seems really odd to me to have this vanishingly small class...   Could this be made into part of the `AvatarCacheSource`?  Should it?

Reply:
Hmm. I see what you are saying, but after thinking about it, I think it may make sense to leave it alone. `AvatarCacheSource` is an interface intended to be implemented by other plugins - and this functionality is an internal part of AvatarCache, that could possibly change in the future. We could, of course, provide an abstract class instead of an interface, but my gut feeling is we should keep it simple and not mix the two.

Reply:
I'll let you resolve this one, if you are satisfied :-)